reference/src/client/src/sa_engine_internal.h

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2022 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#ifndef SA_ENGINE_INTERNAL_H
+#define SA_ENGINE_INTERNAL_H
+
+#include "sa_common.h"
+#include <openssl/engine.h>
+#include <threads.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define BLOCK_SIZE_aes_cbc 16
```

Comment:
It makes the DECLARE_CIPHER macro work. It mainly has to do with the OpenSSL constant NID_aes_128_cbc and others. The aes and cbc have to be the same case as what's in that constant name. Otherwise the macro gets more complicated in needing both the lower case and uppercase spelling of AES, CBC, CTR, ECB, and GCM. I can comment the reasoning on the define.

Reply:
Done